vcd/resource_vcd_external_network.go

```diff
@@ -0,0 +1,334 @@
+				ForceNew:    false,
+				Description: "",
+			},
+		},
+	}
+}
+
+// Creates a new external network from a resource definition
+func resourceVcdExternalNetworkCreate(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[TRACE] external network creation initiated")
+
+	vcdClient := meta.(*VCDClient)
+
+	// external network creation is accessible only in administrator API part
+	_, err := vcdClient.GetAdminOrgFromResource(d)
+	if err != nil {
+		return fmt.Errorf(errorRetrievingOrg, err)
+	}
+
+	params, err := getExternalNetworkInput(d, vcdClient)
+	if err != nil {
+		return err
+	}
+
+	task, err := govcd.CreateExternalNetwork(vcdClient.VCDClient, params)
+	if err != nil {
+		log.Printf("[DEBUG] Error creating external network: %#v", err)
+		return fmt.Errorf("error creating external network: %#v", err)
+	}
+
+	err = task.WaitTaskCompletion()
+	if err != nil {
+		log.Printf("[DEBUG] Error waiting for external network to finish: %#v", err)
+		return fmt.Errorf("error waiting for external network to finish: %#v", err)
+	}
+
+	d.SetId(d.Get("name").(string))
+	log.Printf("[TRACE] external network created: %#v", task)
+	return resourceVcdExternalNetworkRead(d, meta)
+}
+
+// Fetches information about an existing external network for a data definition
+func resourceVcdExternalNetworkRead(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[TRACE] external network read initiated")
+
+	vcdClient := meta.(*VCDClient)
+
+	externalNetworkRef, err := govcd.GetExternalNetworkByName(vcdClient.VCDClient, d.Id())
+	if err != nil || *externalNetworkRef == (types.ExternalNetworkReference{}) {
+		log.Printf("[DEBUG] Unable to find external network. Removing from tfstate")
+		d.SetId("")
+		return nil
+	}
+
+	externalNetwork := govcd.NewExternalNetwork(&vcdClient.VCDClient.Client)
+	externalNetwork.ExternalNetwork.HREF = externalNetworkRef.HREF
+	err = externalNetwork.Refresh()
+	if err != nil {
+		log.Printf("[DEBUG] Error fetching external network details %#v", err)
+		return fmt.Errorf("error fetching external network details %#v", err)
+	}
+
+	log.Printf("[TRACE] external network read completed: %#v", externalNetwork)
+	return nil
+}
+
+//update function for "delete_force", "delete_recursive" no actions needed
+func resourceVcdExternalNetworkUpdate(d *schema.ResourceData, m interface{}) error {
+	return nil
+}
+
+// Deletes a external network, optionally removing all objects in it as well
+func resourceVcdExternalNetworkDelete(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[TRACE] external network delete started")
+
+	vcdClient := meta.(*VCDClient)
+
+	externalNetworkRef, err := govcd.GetExternalNetworkByName(vcdClient.VCDClient, d.Id())
+	if err != nil || *externalNetworkRef == (types.ExternalNetworkReference{}) {
+		log.Printf("[DEBUG] Unable to find external network. Removing from tfstate")
+		d.SetId("")
+		return nil
+	}
+
+	externalNetwork := govcd.NewExternalNetwork(&vcdClient.VCDClient.Client)
```

Comment:
If `govcd.GetExternalNetworkByName` would return external network struct, then this part would be more streamline. I will add request in other PR.